scripts/entity/world/settlements/buildings/stronghold_storage_building.nut

```diff
@@ -160,109 +160,131 @@ this.stronghold_storage_building <- this.inherit("scripts/entity/world/settlemen
-					medicine_count++
-				}
-				if(ammo_count < ammo_max){
-					toAdd.push(this.new("scripts/items/supplies/ammo_item"));
-					ammo_count++
-				}
-				
+				moneyItem = item;
+				continue;
 			}
-			for (local y = 0; y < tool_increment; y++){
-				if(tool_count < tool_max){
-					toAdd.push(this.new("scripts/items/supplies/armor_parts_item"));
-					tool_count++
+			foreach(itemToAdd in itemsToAdd)
+			{
+				if (itemToAdd.ID == item.m.ID)
+				{
+					itemToAdd.ToAdd += item.getAmount();
+					toRemove.push(item);
+					break;
 				}
 			}
 		}
-		if (this.m.Settlement.hasAttachedLocation("attached_location.gold_mine"))
+		// Remove current items and just go off total. Can't remove items from array while iterating through it so do that here
+		foreach(item in toRemove)
 		{
-			local money = this.new("scripts/items/supplies/money_item");
-			local moneyPerMine = this.Stronghold.DailyGoldPerGoldMine;
-			if (this.Stronghold.AllowMoreThanOneGoldMine) {
-				// More than one possible mine scenario, money = number of mines * daily money per mine * number of days
-				// Get number of mines
-				local numberOfMines = countAttachedLocations("attached_location.gold_mine");
-				money.setAmount(_daysPassed * numberOfMines * moneyPerMine);
-			} else {
-				// Only one possible mine scenario, money = daily money per mine * number of days
-				money.setAmount(_daysPassed * moneyPerMine);
-			}
-			// Now that money value is calculated, add it to stash
-			this.m.Stash.add(money);
+			this.m.Stash.remove(item);
 		}
 
-		// Finally, add all items to be added to the stash/marketplace
-		foreach (item in toAdd)
+		local item;
+		local amount;
+		foreach(itemToAdd in itemsToAdd)
 		{
-			item.m.PriceMult = 0;
-			this.m.Stash.add(item);
+			// can't add more than storage max of each item in total
+			itemToAdd.ToAdd = this.Math.min(itemToAdd.ToAdd, itemToAdd.Max);
+			while (itemToAdd.ToAdd > 0)
+			{
+				item = this.new(itemToAdd.Script);
+				amount = this.Math.min(itemToAdd.MaxPerStack, itemToAdd.ToAdd).tointeger();
+				item.setAmount(amount);
+				itemToAdd.ToAdd -= amount;
+				item.m.PriceMult = 0;
+				this.m.Stash.add(item);
+			}
 		}
-	}
-
-	function countAttachedLocations( _id )
-	{
-		local count = 0;
-		foreach( a in this.m.Settlement.getActiveAttachedLocations() )
+		
+		if (this.m.Settlement.hasAttachedLocation("attached_location.gold_mine"))
 		{
-			if (a.isActive() && a.getTypeID() == _id)
+			local moneyPerMine = this.Stronghold.Locations["Gold_Mine"].DailyIncome;
+			local numberOfMines = this.m.Settlement.countAttachedLocations("attached_location.gold_mine");
+			local totalMoney = _daysPassed * numberOfMines * moneyPerMine;
+			if(moneyItem == null)
+			{
+				moneyItem = this.new("scripts/items/supplies/money_item");
+				moneyItem.setAmount(totalMoney);
+				this.m.Stash.add(moneyItem);
+			}
+			else
 			{
-				count = count + 1;
+				moneyItem.setAmount(moneyItem.getAmount() + totalMoney);
```

Comment:
can a moneyItem overflow? what's the max amount it can hold?

because i've seen stashed money items either shrink in size or just overflow.

Reply:
No idea, guess I'll test it.